opencensus/tags/tag_key.py

```diff
@@ -12,36 +12,23 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from opencensus.tags.validation import is_valid_tag_name
 
-class TagKey(object):
-    """ A tag key with a property name
+_TAG_NAME_ERROR = \
+    'tag name must not be empty,' \
+    'no longer than 255 characters and of ascii values between 32 - 126'
 
-    :type name: str
-    :param name: The name of the key
 
-    """
-    def __init__(self, name):
-        self._name = name
+class TagKey(str):
+    """A tag key with a property name"""
 
-    @property
-    def name(self):
```

Comment:
Why remove this?

Reply:
Seems like an overkill.
Using `property` was unnecessary, since we can just use `name` by itself.

also, in other implementations eventually `TagKey` is an object with string value, so inheriting from `str` enables us all its benefits while keeping it simple.

can keep the `name` field for consistency across languages thought it's not specified in the specs